src/main/java/com/rabbitmq/perf/CommandLineProxy.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2018-Present Pivotal Software, Inc.  All rights reserved.
+//
+// This software, the RabbitMQ Java client library, is triple-licensed under the
+// Mozilla Public License 1.1 ("MPL"), the GNU General Public License version 2
+// ("GPL") and the Apache License version 2 ("ASL"). For the MPL, please see
+// LICENSE-MPL-RabbitMQ. For the GPL, please see LICENSE-GPL2.  For the ASL,
+// please see LICENSE-APACHE2.
+//
+// This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY KIND,
+// either express or implied. See the LICENSE file for specific language governing
+// rights and limitations of this software.
+//
+// If you have any questions regarding licensing, please contact us at
+// info@rabbitmq.com.
+
+package com.rabbitmq.perf;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.lang.String.valueOf;
+
+/**
+ * A proxy to add behavior around a {@link CommandLine}.
+ * It implements only the few methods used in {@link PerfTest}.
+ * This class has been introduced to easily make environment
+ * variables override or directly set arguments.
+ *
+ * {@link CommandLine} doesn't implement any interface nor
+ * can be subclassed, that's why this proxy trick is used.
```

Comment:
Just for my own knowledge, why can't [`CommandLine`](https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/CommandLine.html) be subclassed?